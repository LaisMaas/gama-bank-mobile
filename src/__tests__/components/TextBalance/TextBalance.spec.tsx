import React from 'react';
import renderer from 'react-test-renderer';
import TextBalance from '../../../components/TextBalance';

jest.mock("@react-navigation/native", () => {
    const actualNav = jest.requireActual("@react-navigation/native")
    return {
      ...actualNav,
      useFocusEffect: () => jest.fn(),
      useNavigation: () => ({
        navigate: jest.fn(),
      }),
    }
  })

jest.mock('react-native-gesture-handler', () => {
  const View = require('react-native/Libraries/Components/View/View');
  return {
    Swipeable: View,
    DrawerLayout: View,
    State: {},
    ScrollView: View,
    Slider: View,
    Switch: View,
    TextInput: View,
    ToolbarAndroid: View,
    ViewPagerAndroid: View,
    DrawerLayoutAndroid: View,
    WebView: View,
    NativeViewGestureHandler: View,
    TapGestureHandler: View,
    FlingGestureHandler: View,
    ForceTouchGestureHandler: View,
    LongPressGestureHandler: View,
    PanGestureHandler: View,
    PinchGestureHandler: View,
    RotationGestureHandler: View,
    /* Buttons */
    RawButton: View,
    BaseButton: View,
    RectButton: View,
    BorderlessButton: View,
    /* Other */
    FlatList: View,
    gestureHandlerRootHOC: jest.fn(),
    Directions: {},
  };
});

describe('<TextBalance />', () => {
  it('Should render <TextBalance />', () => {
    const tree = renderer.create(<TextBalance/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});