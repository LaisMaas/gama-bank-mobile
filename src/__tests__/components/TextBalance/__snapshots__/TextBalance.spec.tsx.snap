// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextBalance /> Should render <TextBalance /> 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#34A6E7",
        "fontSize": 28,
        "fontWeight": "bold",
        "marginTop": 15,
        "textAlign": "center",
      },
    ]
  }
/>
`;
